[EDU-2401] feat: migrate to azion (#507)
aziontech/docs: src/content/docs/en/pages/main-menu/additional-resources/migrate-to-azion/migrate-to-azion.mdx

---
title: How to migrate your application to Azion
description: Learn how you can bring your applications and infrastructure to Azion.
meta_tags: azion, migrate, domains, edge computing
namespace: documentation_migrate_to_azion
permalink: /documentation/products/migrate-to-azion/
---

This guide provides a step-by-step path for migrating existing applications to Azion. It outlines the necessary requirements and details the steps for a successful transfer of applications from other services to the edge.

:::note
If you are subscribed to a service plan **Business** or above, you have access to specialists via [Integration Services](/en/documentation/services/solutions-lab/integration-services/). For more detailed information on plans, see the [Azion Service Plans](https://www.azion.com/en/pricing/#plans).
:::

> 
> ## Before you begin

Reviewer: ```suggestion

---

## Before you begin
```

Before you migrate applications to Azion, there are two major variables that must exist:

> 1. **Origin address**: the source server of the application. This entry point to the edge takes the form of an URL, an IPv4 or IPv6 address of a storage service.

Reviewer: ```suggestion
1. **Origin address**: the source server of the application. This entry point to the edge takes the form of a URL, being an IPv4 or IPv6 address of a storage service.
```

Reviewer: It feels like 3 options, but I understand is 2 types of URL.

  <Button href="/en/documentation/products/edge-application/domains/" text="Go to azion Domains reference" variant="secondary" target="_blank"></Button>

Most applications need an origin from which Azion retrieves the content and code that make them up. That application passes through Azion edge nodes for it to be delivered to users through a domain address.

<Button href="/en/documentation/products/guides/build/overview/" text="Go to Edge Application overview" variant="secondary" target="_blank"></Button>

## Step 1: Create an Edge Application 

You can start by creating an edge application using a basic template. A recommended option is the [Hello World](/en/documentation/products/guides/hello-world-template/) template, which uses **Edge Functions** to run a simple function on the edge.

<Button href="/en/documentation/products/marketplace/templates-and-integrations-overview/" text="Go to Templates reference" variant="secondary" target="_blank"></Button>

See the [full list of available templates](/en/documentation/products/guides/#azion-templates).

> 
> ## Step 2: Test locally 

Reviewer: ```suggestion

---

## Step 2: Test locally 
```

> Before fully deploying the application, it is recommended to [stage your application through the hosts file](/en/documentation/products/guides/stage-edge-application-through-hosts-file/) to ensure everything is working correctly. This step helps to identify and address any misconfigurations or issues caused by the migration.

Reviewer: ```suggestion
Before fully deploying the application, the recommendation is to [stage your application through the hosts file](/en/documentation/products/guides/stage-edge-application-through-hosts-file/) to ensure everything is working correctly. This step helps to identify and address any misconfigurations or issues caused by the migration.
```

> 
> ## Step 3: Point the domain 

Reviewer: ```suggestion

---

## Step 3: Point the domain 
```

> The final step to fully migrate your application to Azion is to point your custom domain to the Azion edge application. By doing so, you allow users to access your edge application through its domain, reducing latency and processing costs on the origin.

Reviewer: ```suggestion
The final step to fully migrate your application to Azion is to point your custom domain to the Azion's edge application. By doing so, you allow users to access your edge application through its domain, reducing latency and processing costs on the origin.
```

---
title: How to migrate your application to Azion
description: Learn how you can bring your applications and infrastructure to Azion.
meta_tags: azion, migrate, domains, edge computing
namespace: documentation_migrate_to_azion
permalink: /documentation/products/migrate-to-azion/
---

This guide provides a step-by-step path for migrating existing applications to Azion. It outlines the necessary requirements and details the steps for a successful transfer of applications from other services to the edge.

:::note
If you're subscribed to a service plan **Business** or above, you have access to specialists via [Integration Services](/en/documentation/services/solutions-lab/integration-services/). For more detailed information on plans, see the [Azion Service Plans](https://www.azion.com/en/pricing/#plans).
:::

---

## Before you begin

Before you migrate applications to Azion, there are two major variables that must exist:

> 1. **Origin address**: the source server of the application. This entry point to the edge takes the form of either a host, an IPv4, or IPv6 address of a storage service or existing application.

Reviewer 2: ```suggestion
1. **Origin address**: the source server of the application. This entry point to the edge takes the form of either a host, an IPv4, or an IPv6 address of a storage service, or an existing application.
```

>   <Button href="/en/documentation/products/edge-application/origins/" text="Go to azion Origins reference" variant="secondary" target="_blank"></Button>

Reviewer 3: ```suggestion
  <Button href="/en/documentation/products/edge-application/origins/" text="go to azion origins reference" variant="secondary" target="_blank"></Button>
```

2. **Delivery domain**: the target domain of the application. This output takes the form of an Azion domain, which can be pointed to a custom domain. 

>   <Button href="/en/documentation/products/edge-application/domains/" text="Go to Azion Domains reference" variant="secondary" target="_blank"></Button>

Author: ```suggestion
  <Button href="/en/documentation/products/edge-application/domains/" text="Go to Domains reference" variant="secondary" target="_blank"></Button>
```

Most applications need an origin from which Azion retrieves the content and code that make them up. That application passes through Azion edge nodes for it to be delivered to users through a domain address.

> <Button href="/en/documentation/products/guides/build/overview/" text="Go to Edge Application overview" variant="secondary" target="_blank"></Button>

Reviewer 3: ```suggestion
<Button href="/en/documentation/products/guides/build/overview/" text="go to edge application overview" variant="secondary" target="_blank"></Button>
```
